### PR TITLE
Support subdir for [github-check,github-pr-check] and fix subdir handling for other reporter as well

### LINKED ISSUE #225: --reporter=github-pr-review does not work with sub directory
when run go vet and reviewdog in sub directory, reviewdog do nothing.
from my forking & analysis, `-strip` option is working but wd in `FilterCheck()` is not working well.

## info

* go version: 1.12 (`govet` format also not work, I use `-efm "%f:%l:%c: %m`)
* install with install.sh as README
* on CircleCI Enterprise 2.x
* with GitHub Enterprise

## how I did

configured reviewdog run with go vet in `.circleci/config.yml` like:

```
- run:
  command: |
    test ${CI_PULL_REQUEST} || exit 0
    cd subdir
    curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b ./bin
    go vet ./... 2>&1 | ./bin/reviewdog -name "go vet" -efm "%f:%l:%c: %m" -reporter=github-pr-review || true
```
before the commands,  I configured api token and url as environment variable.

when directory structure is like:

```
|- .circleci/config.yml
|- directories including codes with not go codes
|- subdir
    |- go codes...
```

`go vet` reports `unreachable code` which i created for testing reviewdog, and the line is in PR changes like:
スクリーンショット 2019-06-18 17 14 17

## repro example

https://github.com/nasa9084/reviewdog-test/pull/1

I want review comment by reviewdog but not

```
#!/bin/bash -eo pipefail
test ${CI_PULL_REQUEST} || exit 0
cd subdir
curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b ./bin
go vet ./... 2>&1 | ./bin/reviewdog -name "go vet" -efm "%f:%l:%c: %m" -reporter=github-pr-review || true

reviewdog/reviewdog info checking GitHub for latest tag
reviewdog/reviewdog info found version: 0.9.12 for v0.9.12/Linux/x86_64
reviewdog/reviewdog info installed ./bin/reviewdog
```

### LINKED ISSUE #461: [github-pr-check,github-check] Support sub-directory run

### PR DESCRIPTION
**Pull request by haya14busa**
1. Support subdir for [github-check,github-pr-check] (Fix #461)
2. Fix #225

### CODE REVIEW

**Comment by ChadTaljaardt** [comment]
Thanks for this PR @haya14busa

**Comment by ChadTaljaardt** [comment]
What is required for this to get merged?

**Comment by ChadTaljaardt** [comment]
Is there any update on this PR status? Can it be merged?

**Comment by seve** [comment]
Any update on merging this? I would love to be able to use this on my team's project

**Comment by ChadTaljaardt** [comment]
Our team might need to look into a different solution if this doesn't get reviewed soon, please if you get time can you merge this @mgrachev

File: service/commentutil/commentutil.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package commentutil
+
+import (
+	"fmt"
+
+	"github.com/reviewdog/reviewdog"
+)
+
+// `path` to `position`(Lnum for new file) to comment `body`s
```
Comment (review-dog) [inline service/commentutil/commentutil.go:9]: **[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
comment on exported type PostedComments should be of the form "PostedComments ..." (with optional leading article)

**Comment by haya14busa** [comment]
This PR was pending because I noticed that it actually didn't work well for other reporters like `github-pr-review` as well (#225).
I fixed it now, so I can merge it soon.

image

**Pull request merged**